lib: lte_link_control: multiple registered LTE event handlers and better testability (#5719)
nrfconnect/sdk-nrf: doc/nrf/releases/release-notes-changelog.rst

  * :ref:`lte_lc_readme` library:

    * Changed the value of an invalid E-UTRAN cell ID from zero to UINT32_MAX for the LTE_LC_EVT_NEIGHBOR_CELL_MEAS event.

>     * Added a support for multiple LTE event handlers. Thus, deregistation is not anymore possible by using lte_lc_register_handler(NULL), and now on, there is own function in API for doing that: lte_lc_deregister_handler().

Reviewer: ```suggestion
    * Added support for multiple LTE event handlers. Thus, deregistration is not possible by using lte_lc_register_handler(NULL) anymore and it is done by the :c:func:`lte_lc_deregister_handler` function in the API.
```

Author: Thanks @umapraseeda. That change is now done.
